docs/docs/icechunk-python/version-control.md

```diff
@@ -46,55 +45,51 @@ Now that we have a `Repository` with a `main` branch, we can modify the data in
 
     Writable `Session` objects are required to create new snapshots, and can only be created from the tip of a branch. Checking out tags or other snapshots is read-only.
 
-```python
+```python exec="on" session="version" source="material-block"
 session = repo.writable_session("main")
 ```
 
 We can now access the `zarr.Store` from the `Session` and create a new root group. Then we can modify the attributes of the root group and create a new snapshot.
 
-```python
+```python exec="on" session="version" source="material-block" result="code"
 import zarr
 
 root = zarr.group(session.store)
 root.attrs["foo"] = "bar"
-session.commit(message="Add foo attribute to root group")
-
-# 'J1ZJHS4EEQW3ATKMV9TG'
+print(session.commit(message="Add foo attribute to root group"))
 ```
 
 Success! We've created a new snapshot with a new attribute on the root group.
 
 Once we've committed the snapshot, the `Session` will become read-only, and we can no longer modify the data using our existing `Session`. If we want to modify the data again, we need to create a new writable `Session` from the branch. Notice that we don't have to refresh the `Repository` to get the updates from the `main` branch. Instead, the `Repository` will automatically fetch the latest snapshot from the branch when we create a new writable `Session` from it.
 
-```python
+```python exec="on" session="version" source="material-block"
 session = repo.writable_session("main")
 root = zarr.group(session.store)
 root.attrs["foo"] = "baz"
-session.commit(message="Update foo attribute on root group")
-
-# 'BZ9YP38SWPW2E784VAB0'
+print(session.commit(message="Update foo attribute on root group"))
 ```
 
 With a few snapshots committed, we can take a look at the ancestry of the `main` branch:
 
-```python
+
+```python exec="on" session="version" source="material-block" result="code"
 for snapshot in repo.ancestry(branch="main"):
     print(snapshot)
```

Comment:
```suggestion
print(list(repo.ancestry(branch="main"))
```
perhaps?

Reply:
renders poorly unfortunately. As one really long line:

<img width="716" alt="image" src="https://github.com/user-attachments/assets/bb2a1fde-82f7-4b6c-8102-33a7edb1c5f5" />

Reply:
Ah too bad. we should work on a better repr for that ancestry object.